docs/special-classes.md

```diff
@@ -0,0 +1,42 @@
+# Special classes
+
+Some APIs needs to be handled differenlty based on the output language, while others share many common parameters.
+This document contains the list of this special classes and where those should be used.
+
+## CatRequestBase
+
+This class must be used when implementing any Cat request definition.
+
+For example:
+```ts
+@rest_spec_name("cat.aliases")
+class CatAliasesRequest extends CatRequestBase {
+  path_parts?: {
+    name?: Names;
+  }
+  query_parameters?: {
+    expand_wildcards?: ExpandWildcards;
+  }
+  body?: {
+  }
+}
+```
+
+## CatResponseBase
+
+This class must be used when implementing any Cat response definition.
+
+For example:
+```ts
+class CatAliasesResponse extends CatResponseBase<CatAliasesRecord> {}
+```
+
+## HeadResponseBase
```

Comment:
I am not feeling the name `HeadResponseBase` since its too tied to the HTTP method e.g why is there no `PostResponseBase` etcetera. How about `EmptyResponseBase` ?

Reply:
`EmptyResponseBase` could work as well, I liked `HeadResponseBase` because is something that will never leak into language implementations, but it will be used by generators to define the behavior in case of HEAD responses.

Reply:
I am not sure `EmptyResponseBase` won't leak always. I am not sure if we'll expose it in .NET. Currently we already have `e.g` `ExistsResponse` and we might keep that as a parent.